globus_automate_client/cli/callbacks.py

```diff
@@ -114,6 +114,38 @@ def principal_validator_callback(principals: List[str]) -> List[str]:
     return principals
 
 
+def principal_or_all_authenticated_users_validator(principals: List[str]) -> List[str]:
+    """
+    Certain fields expect values to be a valid Globus Auth UUID or one of a set
+    of special strings that are meaningful in the context of authentication.
+    This callback is a specialized form of the principal_validator where the
+    special value of 'all_authenticated_users' is accepted.
+    """
+    special_vals = {"all_authenticated_users"}
```

Comment:
per the above, this kinda becomes `return principal_validator(principals, special_vals={"all_authenticated_users"})`.